corehq/apps/cleanup/tasks.py

```diff
@@ -190,3 +191,21 @@ def check_for_sql_forms_without_existing_domain():
         mail_admins_async.delay(
             'All SQL forms belong to valid domains', ''
         )
+
+
+@periodic_task(run_every=crontab(minute=0, hour=0), queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'))
+def check_for_elasticsearch_data_without_existing_domain():
+    existing_domain_names = set(Domain.get_all_names())
```

Comment:
It might be better to split this up into multiple tasks. 

```python
def check_for_elasticsearch_data_without_existing_domain():
    for domain in all_domains:
        actually_check.delay(domain)

@task
def actually_check(domain):
   pass
```

Reply:
I see your point about smaller tasks / moving them out of periodic, but this wouldn't result in smaller queries, would it?  It's necessary to include all of the current domains in the query so we find any data that doesn't belong to an existing domain.

Reply:
ohh. now i see. We don't actually have a list of domains that have been deleted. 

I think that when we delete a domain we should reserve it's name so that it can never be used again. I that world this would be a necessary, one time check and all future deleted domains won't be able to be reused. Then we could iterate over deleted domains for much smaller checks

Reply:
> I think that when we delete a domain we should reserve it's name so that it can never be used again.

I agree with this.

Reply:
e083bdf008dad22fed7b628ce59c7b78287a6e72

Reply:
I think reserving domain names is a bit out of scope (although it would be good if fully specced out).  I updated the PR to check for only deleted domains.

Reply:
agreed that's out of scope for this. thanks for the follow ups